src/codegen/ast_interpreter.cpp

```diff
@@ -609,6 +630,21 @@ Value ASTInterpreter::visit_invoke(AST_Invoke* node) {
         v = visit_stmt(node->stmt);
         next_block = node->normal_dest;
     } catch (ExcInfo e) {
+
+        if (threading::ThreadStateInternal::getUnwindState() == UNWIND_STATE_NORMAL) {
```

Comment:
It does appear to work, using this (which I'm not sure about):

```
static bool isReraise(AST_stmt* stmt) {                                                                                        
    if (stmt->type != AST_TYPE::Raise) return false;                                                                           
    AST_Raise* raise = static_cast<AST_Raise*>(stmt);                                                                          
    return raise->arg0 == NULL  // the 0-arg form                                                                              
        || raise->arg2 != NULL; // the 3-arg form                                                                              
}                                                                                                                              
```

are invoke statements limited to raises?